moveit_core/robot_model/include/moveit/robot_model/link_model.h

```diff
@@ -41,13 +41,16 @@
 #include <vector>
 #include <utility>
 #include <map>
+#include <boost/shared_ptr.hpp>
 #include <Eigen/Geometry>
 #include <eigen_stl_containers/eigen_stl_vector_container.h>
 #include <moveit/macros/class_forward.h>
 
 namespace shapes
 {
-MOVEIT_CLASS_FORWARD(Shape);
+class Shape;
+typedef boost::shared_ptr<Shape> ShapePtr;
+typedef boost::shared_ptr<const Shape> ShapeConstPtr;
```

Comment:
Nevertheless, ros-planning/geometric_shapes#57 and using `MOVEIT_CLASS_FORWARD` here again will also solve the practical issue.

That would basically mean dropping one commit from this PR:
* Don't use MOVEIT_CLASS_FORWARD for non-moveit classes.